src/wp-admin/includes/image.php

```diff
@@ -291,30 +291,55 @@ function wp_create_image_subsizes( $file, $attachment_id ) {
 		 * If the original image's dimensions are over the threshold,
 		 * scale the image and use it as the "full" size.
 		 */
+		$scale_down = false;
+		$convert    = false;
+
 		if ( $threshold && ( $image_meta['width'] > $threshold || $image_meta['height'] > $threshold ) ) {
+			// The image will be converted if needed on saving.
+			$scale_down = true;
+		} else {
+			// The image may need to be converted regardless of its dimensions.
+			$output_format = wp_get_image_editor_output_format( $file, $imagesize['mime'] );
+
+			if ( is_array( $output_format ) && array_key_exists( $imagesize['mime'], $output_format ) ) {
+				$convert = true;
+			}
+		}
+
+		if ( $scale_down || $convert ) {
```

Comment:
```suggestion
		if ( true === $scale_down || true === $convert ) {
```

We know for sure that these are bool, so why not do a strict comparison here?

Reply:
Both `$scale_down` and `$convert` are designed to be hard-coded boolean variables. They cannot be anything else. Thinking that strict comparison is redundant here, and may perhaps give the impression that these simple vars can be more complex/different types which would be undesirable.